src/unix/apple/macosx/app.m

```diff
@@ -1240,6 +1263,18 @@ void MTY_AppSetPNGCursor(MTY_App *ctx, const void *image, size_t size, uint32_t
 		NSData *data = [NSData dataWithBytes:image length:size];
 		NSImage *nsi = [[NSImage alloc] initWithData:data];
 
+		// Check for cursor scaling hint in the PNG image.
+		float cursor_scale = app_extract_private_scale_tag_from_png_cursor(image, size);
+		if (cursor_scale != 1.0f && [[nsi representations] count] > 0) {
+			NSSize csize = [nsi size];
+			csize.width /= cursor_scale;
+			csize.height /= cursor_scale;
+			nsi.size = csize;
+
+			hotX = (uint32_t) (hotX / cursor_scale + 0.5f);
+			hotY = (uint32_t) (hotY / cursor_scale + 0.5f);
```

Comment:
```suggestion
			hotX = (uint32_t) ((float) hotX / cursor_scale + 0.5f);
			hotY = (uint32_t) ((float) hotY / cursor_scale + 0.5f);
```
(A change originally suggested by Ronald that you probably missed)

Reply:
@ammon-c